sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -274,3 +298,137 @@ private[sql] object OrcFilters {
     }
   }
 }
+
+/**
+ * This case class represents a position in a `Filter` tree, paired with information about
```

Comment:
Seems like using reflection to instantiate `ExpressionTree` is easier than this approach.